Sources/Classes/RSDBPersistentManager.m

```diff
@@ -151,6 +100,29 @@ - (BOOL)isEncryptionNeeded:(RSDBEncryption * __nullable)dbEncryption {
     return NO;
 }
 
+- (BOOL) isSQLCipherAvailable {
+    BOOL isSQLCipherAvailable = NO;
+    @try {
+        void *stmt = nil;
+        if ([database open_v2:[[self getEncryptedDBPath] UTF8String] flags:SQLITE_OPEN_CREATE | SQLITE_OPEN_READWRITE | SQLITE_OPEN_FULLMUTEX zVfs:NULL] == SQLITE_OK && [database prepare_v2:"PRAGMA cipher_version;" nBytes:-1 ppStmt:&stmt pzTail:NULL] == SQLITE_OK) {
+            if ([database step:stmt] == SQLITE_ROW) {
+                const unsigned char *ver = [database column_text:stmt i:0];
+                if(ver != NULL) {
+                    isSQLCipherAvailable = YES;
+                }
+            }
+            [database finalize:stmt];
+        }
+        [self closeDB];
```

Comment:
Don't close & remove the DB here. Do it where you are checking the result.

Reply:
Yes.

Reply:
Can you please help me understand why we need to do it from the place where this method is called ?

Reply:
This method is for checking if SQLCipher is available or not.! This method should not contain other stuff, violating the single responsibility principle.

Reply:
Updated the method to do the SQLCipher availability test on a new database instead of the encrypted or unencrypted databases used for event storage. 

Regarding single responsibility we are aligned on the fact that this method is serving only one responsibility which is to check if SQLCipher is available or not and as part of it we need to connect to DB and run the query and then we close the connection to DB and deleting the DB as it is no longer needed.